### PR TITLE
8343224: print/Dialog/PaperSizeError.java fails with MediaSizeName is not A4: A4

### PR DESCRIPTION
**Pull request by GennadiyKrivoshein**
Fix for https://bugs.openjdk.org/browse/JDK-8343224, print/Dialog/PaperSizeError.java fails with MediaSizeName is not A4: A4.

**Reason of the bug:**
The reason for the regression bug is a missed check for the standard media existence during CustomMediaSizeName creation.
After the JDK-7001133 new MediaSize is created, regardless of whether a standard media size exists. 

For example, if a printer's PPD has two media with the same size, close to "iso_A4," with a small difference.
         *PaperDimension A4/A4:             "595 842"
         *PaperDimension BrA4_B/A4 (Borderless):  "595 842"
After processing the first line, the MediaSize will contain the new size "A4" because of the missed check, and CustomMediaSize will match the "iso_a4" standard media size.
After processing the second line, the MediaSize will contain the new size "BrA4_B", and CustomMediaSize will match not the "iso_a4" but the previously added "A4", because it was added by mistake and its size is exactly the same as "BrA4_B".

**Fix description:**
Create new MediaSize if there is no any other MediaSize with the same size.

**Tests:**
OS: MacOS 12, Linux Ubuntu 22, Windows 11. 
open/test/jdk/java/awt/print/Dialog/PaperSizeError.java
test/jdk/javax/print/CustomMediaSizeNameOOMETest.java
All tests from the test/jdk/javax/print/

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 2 [Reviewers](https://openjdk.org/bylaws#reviewer))

### Issue
 * [JDK-8343224](https://bugs.openjdk.org/browse/JDK-8343224): print/Dialog/PaperSizeError.java fails with MediaSizeName is not A4: A4 (**Bug** - P3)

### Reviewers
 * [Prasanta Sadhukhan](https://openjdk.org/census#psadhukhan) (@prsadhuk - **Reviewer**) Review applies to [987131c6](https://git.openjdk.org/jdk/pull/22429/files/987131c67328895458b2a46b3765b22e800fd2be)
 * [Alisen Chung](https://openjdk.org/census#achung) (@alisenchung - Committer) Review applies to [987131c6](https://git.openjdk.org/jdk/pull/22429/files/987131c67328895458b2a46b3765b22e800fd2be)
 * [Alexander Zvegintsev](https://openjdk.org/census#azvegint) (@azvegint - **Reviewer**)
 * [Alexander Scherbatiy](https://openjdk.org/census#alexsch) (@AlexanderScherbatiy - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22429/head:pull/22429` \
`$ git checkout pull/22429`

Update a local copy of the PR: \
`$ git checkout pull/22429` \
`$ git pull https://git.openjdk.org/jdk.git pull/22429/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22429`

View PR using the GUI difftool: \
`$ git pr show -t 22429`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22429.diff">https://git.openjdk.org/jdk/pull/22429.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22429#issuecomment-2505481663)
</details>

### CODE REVIEW

**Review by prsadhuk — COMMENTED** [review]
Convention is to add the product fix JBS id i.e, this bugid to @bug tag in PaperSizeError.java regression test so that we have a tab of the product fix(es) the test is meant to test..

**Comment by GennadiyKrivoshein** [comment]
> Convention is to add the product fix JBS id i.e, this bugid to @bug tag in PaperSizeError.java regression test so that we have a tab of the product fix(es) the test is meant to test..

Updated.

**Review by prsadhuk — APPROVED** [review]

File: test/jdk/java/awt/print/Dialog/PaperSizeError.java [hunk 1]
```diff
@@ -23,7 +23,7 @@
```
Comment (alisenchung) [inline test/jdk/java/awt/print/Dialog/PaperSizeError.java:23]: could you expand wildcard imports on this test and update copyright year?

Comment (GennadiyKrivoshein) [reply test/jdk/java/awt/print/Dialog/PaperSizeError.java:23]: Done.

**Review by alisenchung — APPROVED** [review]

**Review by prsadhuk — APPROVED** [review]

**Review by azvegint — CHANGES_REQUESTED** [review]
@AlexanderScherbatiy you might want to take a look

File: test/jdk/java/awt/print/Dialog/PaperSizeError.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2007, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
```
[the remark this answers is not in the document]
Comment (GennadiyKrivoshein) [reply test/jdk/java/awt/print/Dialog/PaperSizeError.java]: Thanks, done

**Review by azvegint — APPROVED** [review]

**Review by AlexanderScherbatiy — APPROVED** [review]

**Pull request closed without merging**